### ISSUE TITLE
Ember 3.0 compatibility

### ISSUE THREAD

**Issue by sfbwalder** [albertjan/ember-cli-yadda#56]
ember-cli-yadda is using ember 1.13.x.  It is not compatible with ember 3.x .  This is due to deprecations in 2.x being removed in 3.x.

ember-cli-yadda also needs to embrace new features available in later versions of 2.x and Ember 3.x, including using the Javascript modules API and the new [testing API](https://guides.emberjs.com/v3.0.0/testing/).

**Comment by simonihmig** [comment]
@sfbwalder absolutely!

We are using it in a few apps, and I would like to upgrade things there asap as well. We use it with Mocha, and that needs to be updated to work with the new testing APIs (basically the same update that `ember-qunit` has seen). I have a WIP [PR](https://github.com/emberjs/ember-mocha/pull/190) ready, but this will need to be finalized and released before we can update the test runner code in this repo.

> ember-cli-yadda is using ember 1.13.x. It is not compatible with ember 3.x . This is due to deprecations in 2.x being removed in 3.x.

Curious: what exactly do you see used here, which breaks compatibility?

Looking forward, I see the following action points:
* upgrade to Ember CLI 3.0 using `ember-cli-update` (which will run the `ember-modules-codemod` automatically)
* publish as 0.3.0 (breaking change, as currently node support is still at `0.10` 😬). I think it *should* still be able to run with any 2.x version of Ember/EmberCLI, right?
* Add support for the new testing APIs as soon as upstream support is available (Mocha)

Regarding the latter, I am not entirely sure how we should do this. AFAIK if we opt in into the new APIs, the old test helpers (e.g. global acceptance helpers, `this.$()`) cease to work (not entirely sure though). The thing is that currently there is no way the user can decide which testing APIs to use, as the test runner code by `ember-cli-yadda` generates the setup code implicitly (e.g. [here](https://github.com/albertjan/ember-cli-yadda/blob/master/lib/test-runner/qunit/component.js#L2)). 

So we can either switch to the new testing APIs *only*, and publish that as another breaking change release (0.4.0) and continue to support only that for the future (the previous 0.3.0 release would still be usable for the old APIs), or provide some means for the user to configure this. 

What do you folks think?

I can work on the testing API if nobody else is eager to do this? Would be nice to have the Upgrade ready before that!

**Comment by simonihmig** [comment]
The Mocha PR is merged (https://github.com/emberjs/ember-mocha/pull/190), so we could start with the above steps. 

@albertjan any opinion on the above? Especially this:
> So we can either switch to the new testing APIs only, and publish that as another breaking change release (0.4.0) and continue to support only that for the future (the previous 0.3.0 release would still be usable for the old APIs), or provide some means for the user to configure this.

**Comment by sfbwalder** [comment]
> Curious: what exactly do you see used here, which breaks compatibility?

Perhaps I made an assumption I should not have.  We are using 2.18, and get the following message during the build of our application:
>DEPRECATION: An addon is trying to access project.nodeModulesPath. This is not a reliable way to discover npm modules. Instead, consider doing: require("resolve").sync(something, { basedir: project.root }). Accessed from:   new NPMDependencyVersionChecker (C:\dev\ember\home-ui\node_modules\@sf\ember-cli-yadda\node_modules\ember-cli-version-checker\src\npm-dependency-version-checker.js:11:33)

I assumed when going to 3.0 this would start causing a build failure.  Perhaps it won't.

Good idea to split this into multiple steps.  I seems good to start with an upgrade without changing features.

**Comment by simonihmig** [comment]
> Perhaps I made an assumption I should not have. We are using 2.18, and get the following message during the build of our application

Yeah, I know that message. Pops up quite often, also for projects not related to `ember-cli-yadda`. It seems to be caused by `ember-cli-version-checker`. But what's weird in your error message is that it seems to be a dependency of `ember-cli-yadda` here, while it's actually not used anywhere AFAICT! 😳

But I don't *think* it will break in 3.0. `project.nodeModulesPath` is a private, and now deprecated API of Ember CLI, and will *probably*  be removed somewhere during the 3.x cycle.
